script/bintrayupdate.sh

```diff
@@ -1,8 +1,21 @@
 #!/usr/bin/env bash
+set -e
 # Update the Bintray release
 
+old_date=$(date +"%Y.%m.%d" -d "30 days ago")
 build_dir="$(pwd)/build"
-current_api_url="https://api.bintray.com/packages/streamlink/streamlink-nightly/streamlink/versions/$(date +'%Y.%m.%d')/release_notes"
-latest_api_url="https://api.bintray.com/packages/streamlink/streamlink-nightly/streamlink/versions/latest/release_notes"
-curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" ${current_api_url}
-curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" ${latest_api_url}
+api_base="https://api.bintray.com/packages/streamlink/streamlink-nightly/streamlink"
+current_api_url="${api_base}/versions/$(date +'%Y.%m.%d')/release_notes"
+files_api_url="${api_base}/files"
+versions_api_url="${api_base}/versions"
+
+curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" "${current_api_url}"
+
+echo "Deleting versions older than ${old_date}..."
+versions=$(curl -s -u "${BINTRAY_USER}:${BINTRAY_KEY}" "${files_api_url}" | grep -Po '(?<="version":")[^"]+(?=")')
```

Comment:
Does TravisCI have `jq` installed by default? Then you could replace the (unsafe) regex
`grep -Po '(?<="version":")[^"]+(?=")'`
with a simple
`jq -r '.[].version'`
that returns the version property of all objects in the json array response in a raw format (no quotations)

Reply:
I doubt it, but if there is an ubuntu package it could be added... lemme see

Reply:
Looks like `jq` needs to be added to the apt package list in the travisci config. Package is whitelisted, just checked it.

Reply:
Cool. I read that it's pre-installed, can test it out.

Reply:
`jq` is cool, thanks for the tip